compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1944,6 +1944,9 @@ private Result seekMatchInExpressionRhs(int lookahead, int currentDepth, int cur
             nextContext = ParserRuleContext.MATCH_BODY;
         } else if (parentCtx == ParserRuleContext.MATCH_BODY) {
             nextContext = ParserRuleContext.RIGHT_DOUBLE_ARROW;
+        } else if (parentCtx == ParserRuleContext.TYPE_REFERENCE) {
+            endContext();
+            nextContext = ParserRuleContext.SEMICOLON;
```

Comment:
An expression cannot exist in a `TYPE_REFERENCE` context. Root cause might be somewhere else. Shall we try to find and fix that?

Reply:
I see, I ll try and find that.

Reply:
since we are allowing all type descriptors inside type-reference the error handler goes into expressions as well